### ISSUE TITLE
Dropbox Oauth2 - parameter issues

### ISSUE THREAD

**Issue by nateevans** [MrSwitch/hello.js#301]
When overriding the Dropbox module to use Oauth2, there are a couple issues with parameters being passed:

**1.** The Dropbox api complains about the **state** parameter being too long. Is all the data in **state** required? Any way to cut down on it?

<img src="https://cloud.githubusercontent.com/assets/1561006/9092432/348d8608-3b74-11e5-98ab-49bdc3c99034.png" width="320px" align="top" /> <img src="https://cloud.githubusercontent.com/assets/1561006/9092441/3e34b398-3b74-11e5-83da-5dbf689243dd.png" width="320px"  align="top"/>

**2.** If I get the **state** parameter's length short enough, I run into Dropbox telling me **display** is an unknown parameter. I've tried manually removing the **display** parameter from the query string with no ill effect on both the Dropbox and Google Drive service, can **display** be removed?

<img src="https://cloud.githubusercontent.com/assets/1561006/9093205/a7779fce-3b78-11e5-94cf-68d6e48f6c6e.png" width="320px"  align="top"/>

**Comment by nateevans** [comment]
Actually the full value that's getting passed to dropbox is:

    {"client_id":"tdoi10xxnxga35f","network":"dropbox","display":"popup",
    "callback":"_hellojs_33hqs69x","state":"",
    "redirect_uri":"https://nevansvagrant.ngrok.com/uploadtool/redirect",
    "scope":"email,publish,photos,files,basic",
    "oauth":{"version":2,"auth":"https://www.dropbox.com/1/oauth2/authorize",
    "request":"https://api.dropbox.com/1/oauth/request_token",
    "token":"https://api.dropbox.com/1/oauth/access_token",
    "grant":"https://api.dropbox.com/1/oauth2/token"},
    "oauth_proxy":"https://auth-server.herokuapp.com/proxy"}

Haven't seen where all that gets added yet, so I will keep digging.

**Comment by MrSwitch** [comment]
Hi @nateevans i got this working (see branch `hello-dropbox-oauth2`, although i haven't tested the request length. [The doc says the state can be "Up to 500 bytes of arbitrary data"](https://www.dropbox.com/developers/core/docs#oa2-authorize) and yours above comes to 515 :(

The data is in there because of edge cases on devices where we can't rely on session state. However there are candidates for removing some of that. In the meantime you can shave quite a few bytes by removing the scope, and not using 'Authorisation Code' flow...

```
hello('dropbox').login({
    scope:'',
    response_type:'popup'
});
```

**Comment by MrSwitch** [comment]
Btw: in the oauth object for oauth2, 'request' and 'token' properties dont mean anything.

**Comment by MrSwitch** [comment]
I merged my OAuth2 implementation into v1.8.1 719ed6e

If you have any issues with it please reopen this ticket. Thanks
